### PR TITLE
DFT hessian

### PR DESCRIPTION
**Pull request by ilfreddy**
Linear response calculations need the second derivative of the different energy components.

- [x] Coulomb Hessian
- [x] XC Hessian (LDA)
- [ ] XC Hessian (GGA)
- [ ] Exchange Hessian
- [x] Testing

### CODE REVIEW

File: src/qmoperators/two_electron/CoulombPotential.cpp [hunk 1]
```diff
@@ -84,6 +91,35 @@ void CoulombPotential::setupDensity(double prec) {
...
+    if (this->orbitals == nullptr) MSG_ERROR("Orbitals not initialized");
+    if (this->pertX == nullptr) MSG_ERROR("Perturbed orbitals not initialized");
+
+    OrbitalVector &Phi = *this->orbitals;
+    OrbitalVector &X = *this->pertX;
+    OrbitalVector &Y = *this->pertY;
+    Density &rho = this->perturbedDensity;
+
+    Timer timer;
+    if(this->pertY == nullptr) { // For static perturbations pertY is not required/defined
```
Comment (stigrj) [inline src/qmoperators/two_electron/CoulombPotential.cpp]: Actually, in the static case I think it's set up such that `pertY == pertX`, not `pertY == nullptr`

File: src/qmoperators/two_electron/CoulombOperator.h [hunk 2]
```diff
@@ -15,8 +15,12 @@ namespace mrchem {
 
 class CoulombOperator final : public RankZeroTensorOperator {
 public:
-    CoulombOperator(mrcpp::PoissonOperator *P, OrbitalVector *Phi = nullptr)
-            : potential(P, Phi) {
+ CoulombOperator(mrcpp::PoissonOperator *P,
+                 OrbitalVector *Phi = nullptr,
+                 OrbitalVector *Phi_x= nullptr,
+                 OrbitalVector *Phi_y= nullptr,
+                 int order = 1)
```
Comment (stigrj) [inline src/qmoperators/two_electron/CoulombOperator.h]: How about we skip the `order` argument and make two separate constructors?

File: src/qmoperators/two_electron/CoulombPotential.cpp (hunk 1 above)
Comment (ilfreddy) [reply src/qmoperators/two_electron/CoulombPotential.cpp]: Currently, `pertY` is kept as `nullptr` inside `CoulombPotential` for static perturbations. But I can change it if you prefer.

File: src/properties/Polarizability.h [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+#pragma once
+
+#include "mrchem.h"
+
+namespace mrchem {
+
+class Polarizability final {
+public:
+    Polarizability(double w = 0.0, const double *o = 0, bool v = false) {
```
[the remark this answers is not in the document]
Comment (ilfreddy) [reply src/properties/Polarizability.h]: Done

File: src/chemistry/Molecule.cpp [hunk 4]
```diff
@@ -431,19 +426,16 @@ SpinSpinCoupling& Molecule::getSpinSpinCoupling(int k, int l) {
 
 /** @brief Return property Polarizability */
 Polarizability& Molecule::getPolarizability(double omega) {
-    NOT_IMPLEMENTED_ABORT;
-    /*
     Polarizability *pol_w = 0;
     for (int i = 0; i < this->polarizability.size(); i++) {
         double omega_i = this->polarizability[i]->getFrequency();
-        if (fabs(omega_i - omega) < MachineZero) {
+        if (fabs(omega_i - omega) < mrcpp::MachineZero) {
```
[the remark this answers is not in the document]
Comment (ilfreddy) [reply src/chemistry/Molecule.cpp]: Done

**Pull request closed without merging**